tests/arch/forte_c_test.cpp

```diff
@@ -0,0 +1,87 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Jose Cabral
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Jose Cabral - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+#define BOOST_TEST_MODULE FORTE_C_TEST
+#include <boost/test/included/unit_test.hpp>
+
+#include "../../src/arch/c_interface/forte_c.h"
+
+#include <chrono>
+#include <thread>
+
+BOOST_AUTO_TEST_CASE(forte_c_interface)
+{
+  TForteInstance instance{nullptr};
+
+  // forteStartInstance
+  // invalid instance result
+  BOOST_CHECK_EQUAL(forteStartInstance(61499, nullptr), FORTE_WRONG_PARAMETERS);
```

Comment:
Could you please use `BOOST_TEST` instead of `BOOST_CHECK`, as it results in better test failure output and is nicer to read (at least in my opinion)

Reply:
How does the better output look like? In my opinion, if this test fails you get the output of the actual vs expected value which helps you understand why the test failed, which I believe you don't have if you just use `BOOST_TEST`

Reply:
true, but I get the exact same output. If  I have invalid cases as in:

```
    BOOST_CHECK_EQUAL(forteGlobalInitialize(validArgc, validArgV), 1);
    BOOST_TEST(forteGlobalInitialize(validArgc, validArgV) == 1);
```

I get 

```
~/4diac-forte/tests/arch/forte_c_test.cpp(56): error: in "forte_c_interface": check forteGlobalInitialize(validArgc, validArgV) == 1 has failed [0 != 1]
~/4diac-forte/tests/arch/forte_c_test.cpp(57): error: in "forte_c_interface": check forteGlobalInitialize(validArgc, validArgV) == 1 has failed [0 != 1]
```

maybe I missed something, but I don't see any better output

Reply:
In this simple case, the output seems to be the same, but in my personal experience of writing tests in the past, i found BOOST_TEST generating better test output, but your mileage may vary.

Additionally as mentioned in my first comment about this, one of my reasons to stick to `BOOST_TEST` is the better readability of the check itself.
Just to explain my reasoning, with `BOOST_CHECK_EQUAL` I first get the information the parameters are to be compared for equality, and only after I analyse the parameters (assuming reading from left to right), needing to hold the information of what kind of comparison needs to be performed in my head.
For `BOOST_CHECK_GT` etc. its even worse, as I need to know the order of operators and keep them in mind while checking the parameter expression.

On the other hand, `BOOST_TEST` let you write the expression to be checked in native C/C++ and I can see them in "natural" order of the language.
In addition, I find it easier to only need to know (usually) one test macro, and ignoring it as code noise, focusing on the expression itself.

Also as said above, this is subject to personal preference, and open to discussion of course.
We should also write the results down into our coding guidelines, so the consensus is documented and publicly available in the future.

Reply:
There is a difference when using arithmetic expressions. Please compare the output for:
```c++
int a = 3;
int b = 2;
BOOST_TEST(a + 1 == b);
BOOST_CHECK_EQUAL(a + 1, b);
```
which prints:
```console
test.cpp(8): error: in "test_macro_overview": check a + 1 == b has failed [3 + 1 != 2]
test.cpp(9): error: in "test_macro_overview": check a + 1 == b has failed [4 != 2]
```

> On the other hand, `BOOST_TEST` let you write the expression to be checked in native C/C++ and I can see them in "natural" order of the language. In addition, I find it easier to only need to know (usually) one test macro, and ignoring it as code noise, focusing on the expression itself.

I also personally prefer the ability to write arbitrary conditions in `BOOST_TEST` and let boost figure it out, without having to worry about which test function to use. I find this particularly useful when copy-pasting existing conditions from somewhere.

> Also as said above, this is subject to personal preference, and open to discussion of course. We should also write the results down into our coding guidelines, so the consensus is documented and publicly available in the future.

I would like to add, that according to the boost sources, the `BOOST_CHECK*` macros seem to be deprecated, see
https://github.com/boostorg/test/blob/b0127f48d23b94387c9e57b6b481f9119a3a25db/include/boost/test/tools/interface.hpp#L273-L308. When disabling the "old tools" by defining `BOOST_TEST_NO_OLD_TOOLS`, they simply forward to `BOOST_TEST_CHECK` (as shown in the linked snippet and the output above then becomes identical).

Reply:
> I would like to add, that according to the boost sources, the BOOST_CHECK* macros seem to be deprecated, see
> https://github.com/boostorg/test/blob/b0127f48d23b94387c9e57b6b481f9119a3a25db/include/boost/test/tools/interface.hpp#L273-L308. When disabling the "old tools" by defining BOOST_TEST_NO_OLD_TOOLS, they simply forward to BOOST_TEST_CHECK (as shown in the linked snippet and the output above then becomes identical).

Thanks for finding this, I thought I have read that BOOST_CHECK* have been deprecated, but couldn't find that anymore

Reply:
very interesting. indeed the output is better. I'll change all to `BOOST_TEST`

Reply:
done. One thing that I realized while changing now to `BOOST_TEST` is that you need the `<<` operator for the type you're comparing. 

For example, 

` BOOST_TEST(EMGMResponse::Ready == resource->writeValue(command.mFirstParam, "1", false));`
 does not compile and it requires 

```
std::ostream& operator<<(std::ostream& os, const EMGMResponse& val ){
  // ...  
 return os;
}
```

Reply:
Boost.test gives you the possibility to add a helper function (https://github.com/eclipse-4diac/4diac-forte/blob/develop/tests/forte_boost_output_support.h) instead of the stream operator, in cases where it doesn't make sense for the type itself.

Reply:
wow! I'm completely out of touch regarding `boost`. I updated accordingly